### PR TITLE
refactor: Add PluginFormatter

### PR DESCRIPTION
**Pull request by tienvx**
* Add `MatchingField` matcher
* Add `PluginFormatter`
* Allow customize formatter for `Matcher` helper class
* Apply it to plugin examples e.g. Replace `"matching(type,'Name')"` by `$matcher->like('Name')`
* Syntax and supported matchers of `matching` expression: https://github.com/pact-foundation/pact-plugins/blob/main/docs/matching-rule-definition-expressions.md
* All matchers: https://github.com/pact-foundation/pact-specification/tree/version-4#supported-matching-rules

### CODE REVIEW

**Comment by Lewiscowles1986** [comment]
So one concern with these works, is that anyone using your work, might see backwards compatibility breaks. I Don't think this should be blocked entirely, but I do know that at some point, you'll want a stable interface for pact users and plugin authors alike.

Do you have some form of view around the time-frame, and might it be advantageous to use some form of phpDoc comment to annotate?

**Comment by tienvx** [comment]
A remind for @Lewiscowles1986

**Comment by Lewiscowles1986** [comment]
Ah, I'd missed the reply here. 

RE:

> > Do you have some form of view around the time-frame, and might it be advantageous to use some form of phpDoc comment to annotate?

> Can you rephrase this question for me.

Do you know how long it will take for this to come out of alpha?

Really as you've quantified one user impact, it should not matter anyway.

**Review by Lewiscowles1986 — APPROVED** [review]

**Comment by tienvx** [comment]
> Do you know how long it will take for this to come out of alpha?

I have some items in my mind for the next 2 or 3 more alpha versions:
* Refactor code related to InteractionBuilder/MessageBuilder if needed
* Add unit tests as much as possible
* Update document/readme

Probably it will take 1 or 2 months
